### PR TITLE
:art: Use forever and wait-on

### PR DESCRIPTION
**Pull request by lboyette-okta**
Resolves: OKTA-120043

@rchild-okta 
@haishengwu-okta

### CODE REVIEW

File: tools/daemonUtil.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+const forever = require('forever-monitor');
```
Comment (rchild-okta) [inline tools/daemonUtil.js]: nit: how about just add the .Monitor here, and then below we can just do a normal new Monitor(stuff)

File: tools/daemonUtil.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+daemonUtil.startAppServer = () => {
+  const script = 'start';
+  const color = 'green';
+  return startNpmScript(script, color)
+    .then(child => waitOnPromise({
+      resources: [
+        `http://localhost:${samplesConfig.server.port}/assets/bundle.js`
+      ]
+    }, { script, color })
+      .then(() => child));
```
Comment (rchild-okta) [inline tools/daemonUtil.js]: nit: spacing (here and below)

**Comment by haishengwu-okta** [comment]
This is pretty nice!
shall you promote changes to the generator?

File: tools/daemonUtil.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  return new Promise((resolve, reject) => {
+    child.on('start', () => {
+      resolve(child);
+    });
+    console.log(normal(`Running "npm run ${script}"`));
+    child.start();
+  });
+}
+
+function waitOnPromise(opts, context) {
```
Comment (rchild-okta) [inline tools/daemonUtil.js]: Nit - seems like the below code is all doing the same thing - could probably make this a "runAndWait" type function that just removes some of the boilerplate below.

File: .samples.config.json [hunk 4]
```diff
@@ -8,7 +8,6 @@
     },
     "server": {
       "staticDir": "dist",
-      "startSignal": "Express server started",
```
Comment (rchild-okta) [inline .samples.config.json]: Reminder when updating the generator to also remove the stuff about this in the readme.

**Comment by lboyette-okta** [comment]
@haishengwu-okta I will! I wanted to make sure it passed all the checks here before committing it there.

File: tools/daemonUtil.js (hunk 3 above)
Comment (lboyette-okta) [inline tools/daemonUtil.js]: fair enough. 3 is enough to consider the extra abstraction

File: tools/daemonUtil.js (hunk 1 above)
Comment (lboyette-okta) [inline tools/daemonUtil.js]: makes sense

**Pull request merged**